src/PathDispatcher.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare(strict_types=1);
+
+namespace Northwoods\Broker;
+
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface as Handler;
+
+class PathDispatcher implements Middleware
+{
+    /** @var array */
+    private $middlewares;
+
+    /** @var MiddlewareResolver */
+    private $resolver;
+
+    public function __construct(array $middlewares, MiddlewareResolver $resolver)
+    {
+        $this->middlewares = $middlewares;
+        $this->resolver = $resolver;
+
+        // Make sure the longest path prefixes are matched first
+        // (otherwise, a path /foo would always match, even when /foo/bar
+        // should match).
+        krsort($this->middlewares);
+    }
+
+    public function process(Request $request, Handler $handler): Response
+    {
+        $requestPath = $this->getNormalizedPath($request);
+
+        foreach ($this->middlewares as $pathPrefix => $middleware) {
+            if (strpos($requestPath, $pathPrefix) === 0) {
+                return $this->resolver->resolve($middleware)
+                    ->process(
+                        $this->unprefixedRequest($request, $pathPrefix),
+                        $this->prefixedHandler($handler, $pathPrefix)
+                    );
+            }
+        }
+
+        return $handler->handle($request);
+    }
+
+    private function unprefixedRequest(Request $request, $prefix): Request
+    {
+        $uri = $request->getUri();
+        return $request->withUri(
+            $uri->withPath(
+                substr($uri->getPath(), strlen($prefix))
+            )
+        );
```

Comment:
@shadowhand How would you test this method works correctly, without passing in a real `Request` instance?